--- ./arch/arm/boot/dts/bcm2837-rpi-3-b.dts.orig	2022-01-12 22:31:00.857222036 +0100
+++ ./arch/arm/boot/dts/bcm2837-rpi-3-b.dts	2022-01-12 22:48:45.003928434 +0100
@@ -11,8 +11,16 @@
 	model = "Raspberry Pi 3 Model B";
 
 	chosen {
-		/* 8250 auxiliary UART instead of pl011 */
-		stdout-path = "serial1:115200n8";
+		stdout-path = "";
+		bootargs = "earlyprintk console=tty1";
+	};
+
+	aliases {
+		/* required for the firmware as per
+		 * https://github.com/raspberrypi/firmware/issues/752#issuecomment-283331328
+		 * */
+		uart0 = &uart0;
+		uart1 = &uart1;
 	};
 
 	memory@0 {
@@ -27,10 +35,6 @@
 	};
 };
 
-&bt {
-	shutdown-gpios = <&expgpio 0 GPIO_ACTIVE_HIGH>;
-};
-
 &firmware {
 	expgpio: gpio {
 		compatible = "raspberrypi,firmware-gpio";
@@ -130,17 +134,16 @@
 	status = "okay";
 };
 
-/* uart0 communicates with the BT module */
+/* uart0 is mapped to the pin header */
 &uart0 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&uart0_gpio32 &gpclk2_gpio43>;
+	pinctrl-0 = <&uart0_gpio14>;
+	status = "okay";
 };
 
-/* uart1 is mapped to the pin header */
+/* uart1 is disabled */
 &uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart1_gpio14>;
-	status = "okay";
+	status = "disabled";
 };
 
 /* SDHOST is used to drive the SD card */
