--- ./arch/arm/boot/dts/bcm2837-rpi-3-b-plus.dts.orig	2022-01-11 08:56:41.435321858 +0100
+++ ./arch/arm/boot/dts/bcm2837-rpi-3-b-plus.dts	2022-01-11 20:05:01.466755142 +0100
@@ -11,8 +11,16 @@
 	model = "Raspberry Pi 3 Model B+";
 
 	chosen {
-		/* 8250 auxiliary UART instead of pl011 */
-		stdout-path = "serial1:115200n8";
+		stdout-path = "";
+		bootargs = "earlyprintk console=tty1";
+	};
+
+	aliases {
+		/* required for the firmware as per
+		 * https://github.com/raspberrypi/firmware/issues/752#issuecomment-283331328
+		 * */
+		uart0 = &uart0;
+		uart1 = &uart1;
 	};
 
 	memory@0 {
@@ -145,17 +153,16 @@
 	bus-width = <4>;
 };
 
-/* uart0 communicates with the BT module */
+/* uart0 is mapped to the pin header */
 &uart0 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&uart0_ctsrts_gpio30 &uart0_gpio32 &gpclk2_gpio43>;
+	pinctrl-0 = <&uart0_gpio14>;
+	status = "okay";
 };
 
-/* uart1 is mapped to the pin header */
+/* uart1 is disabled */
 &uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart1_gpio14>;
-	status = "okay";
+	status = "disabled";
 };
 
 &wifi_pwrseq {
