--- ./arch/arm/boot/dts/bcm2711-rpi-4-b.dts.orig	2022-01-12 22:10:16.147145347 +0100
+++ ./arch/arm/boot/dts/bcm2711-rpi-4-b.dts	2022-01-12 22:48:00.869763575 +0100
@@ -10,8 +10,16 @@
 	model = "Raspberry Pi 4 Model B";
 
 	chosen {
-		/* 8250 auxiliary UART instead of pl011 */
-		stdout-path = "serial1:115200n8";
+		stdout-path = "";
+		bootargs = "earlyprintk console=tty1";
+	};
+
+	aliases {
+		/* required for the firmware as per
+		 * https://github.com/raspberrypi/firmware/issues/752#issuecomment-283331328
+		 * */
+		uart0 = &uart0;
+		uart1 = &uart1;
 	};
 
 	leds {
@@ -52,10 +60,6 @@
 	};
 };
 
-&bt {
-	shutdown-gpios = <&expgpio 0 GPIO_ACTIVE_HIGH>;
-};
-
 &ddc0 {
 	status = "okay";
 };
@@ -215,18 +219,16 @@
 	};
 };
 
-/* uart0 communicates with the BT module */
+/* uart0 is mapped to the pin header */
 &uart0 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&uart0_ctsrts_gpio30 &uart0_gpio32>;
-	uart-has-rtscts;
+	pinctrl-0 = <&uart0_gpio14>;
+	status = "okay";
 };
 
-/* uart1 is mapped to the pin header */
+/* uart1 is disabled */
 &uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart1_gpio14>;
-	status = "okay";
+	status = "disabled";
 };
 
 &vc4 {
